interfaces/repo.go

```diff
@@ -550,7 +550,17 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]ConnRef, erro
 			conns = append(conns, connRef)
 		}
 	}
-	return conns, nil
+	// remove duplicates, this is caused by duplicate plug/slot name on core
+	// and missing validation on plug/snap name when adding core.
```

Comment:
Validation of names is unrelated to this problem.

Then, rather than building a list with duplications and then building a map to then build a list without duplications, we can do it at once above by just skipping entries already seen in the first place.